### ISSUE TITLE
Confusion about label and valid dataset.

### ISSUE THREAD

**Issue by vMagina** [lzx551402/contextdesc#21]
There is a huge gap between my train loss and your train loss.
Here is the loss curve.

![1](https://user-images.githubusercontent.com/34497535/76309568-00a62c80-6308-11ea-802a-5244889323fc.jpg)
![3](https://user-images.githubusercontent.com/34497535/76309777-727e7600-6308-11ea-9b2b-a23f4405175e.jpg)

![2](https://user-images.githubusercontent.com/34497535/76309637-22071880-6308-11ea-8aaa-cf41f404b42c.jpg)

- Because my loss is up to 7,000 which is so huge. During my training stage, I use mean operation to replace the sum operation of the Npair loss. I have checked out my network and loss function.But I can't locate the problem. So, can you give me some help about the loss?

- As show in your paper, the loss decay quickly. So, the loss is train loss? Your batch_size is 2. Then, after how much steps you evaluate the model on the valid dataset?

- Besides, do you use the keypoint pair label when caculate the Npair loss to make the product distance between the nosiy keypoint pair slow?

**Comment by vMagina** [comment]
I have corrected my fault.
Now, I use mean loss in the Npair loss and use keypoint labels to only caculate matchable keypoints in the loss.
Then, I get the result: 68.29/75.35 for i/v sequences. After I perturb patches, I get the result: 67.74/75.66 for i/v sequences. So, I think that perturbing on patches may have a little effection on the model. I think the gap between my result and the result in the paper mainly decided by the train dataset. Because I don't know whese image pairs you used. Can you give me some help?

**Comment by lzx551402** [comment]
Hi, you may now refer to [this](https://github.com/lzx551402/tfmatch/blob/master/preprocess.py#L54) on how we sample the image pairs for training.
